src/implementation.cpp

```diff
@@ -687,36 +743,78 @@ simdutf_warn_unused result convert_utf32_to_utf8_with_errors(const char32_t * bu
 simdutf_warn_unused size_t convert_valid_utf32_to_utf8(const char32_t * buf, size_t len, char* utf8_buffer) noexcept {
   return active_implementation->convert_valid_utf32_to_utf8(buf, len, utf8_buffer);
 }
+simdutf_warn_unused size_t convert_utf32_to_utf16(const char32_t * buf, size_t len, char16_t* utf16_buffer) noexcept {
+  #if SIMDUTF_IS_BIG_ENDIAN
+  return active_implementation->convert_utf32_to_utf16be(buf, len, utf16_buffer);
+  #else
+  return active_implementation->convert_utf32_to_utf16le(buf, len, utf16_buffer);
+  #endif
+}
 simdutf_warn_unused size_t convert_utf32_to_utf16le(const char32_t * buf, size_t len, char16_t* utf16_buffer) noexcept {
   return active_implementation->convert_utf32_to_utf16le(buf, len, utf16_buffer);
 }
 simdutf_warn_unused size_t convert_utf32_to_utf16be(const char32_t * buf, size_t len, char16_t* utf16_buffer) noexcept {
   return active_implementation->convert_utf32_to_utf16be(buf, len, utf16_buffer);
 }
+simdutf_warn_unused result convert_utf32_to_utf16_with_errors(const char32_t * buf, size_t len, char16_t* utf16_buffer) noexcept {
```

Comment:
This construction is fine but could be even simpler!!!
```C++
 #if SIMDUTF_IS_BIG_ENDIAN
   return convert_utf32_to_utf16be_with_errors(buf, len, utf16_buffer);
   #else
   return convert_utf32_to_utf16le_with_errors(buf, len, utf16_buffer);
   #endif
```

Reply:
(It is not a request to change your code.)

Reply:
You're right. I'll change them.